src/fftpack_dct.f90

```diff
@@ -2,10 +2,11 @@
 
 contains
 
-    !> Discrete fourier cosine transform of an even sequence.
-    pure module function dct_rk(x, n) result(result)
+    !> Discrete cosine transforms of types 1, 2, 3.
+    pure module function dct_rk(x, n, t) result(result)
```

Comment:
```suggestion
    pure module function dct_rk(x, n, type) result(result)
```
Is it more intuitive for `qct` to use `type` instead of `t` as the argument name and keep similar to [the `scipy.fftpack.dct` interface](https://docs.scipy.org/doc/scipy/reference/generated/scipy.fftpack.dct.html#scipy-fftpack-dct)?

Reply:
`type` was my first choice for this too, but then I thought that I should maybe avoid it because it's also a Fortran keyword. I'm fine with changing back to the more intuitive `type` if you think it's a better choice overall.

Reply:
Because Fortran doesn't reserve keywords, it's okay and appropriate to use `type`, e.g. `result` is also a keyword, but we may use it as the name of the function's return value. :)

Reply:
Understood, I'll make  the change to the more intuitive name `type`.